### ISSUE TITLE
A D105 is raised for a magic method despite ignore = D105

### ISSUE THREAD

**Issue by AlexArcPy** [PyCQA/pydocstyle#362]
Having this piece of code in `bla.py`:

```python
class Foo():
    def __call__(self):
        pass
```

and `flake8`:

> flake8 --version
> 3.7.7 (flake8-comprehensions: 1.4.1, flake8-docstrings: 1.3.0, pydocstyle: 3.0.0, flake8-print: 3.1.0, flake8-string-format: 0.2.3, flake8_builtins: 1.4.1, flake8_commas: 2.0.0, logging-format: 0.6.0, mccabe: 0.6.1, naming: 0.8.2, pycodestyle: 2.5.0, pyflakes: 2.1.0) CPython 2.7.15 on Windows
> 

Will raise 

```
.\bla.py:2:1: D102 Missing docstring in public method
```

Despite having in `setup.cfg`

```
[flake8]
ignore = D105,D100,D101,W292
```

I assume you don't consider the `__call__` to be a magic method

So, `D105 | Missing docstring in magic method` won't be applying to `__call__`. 

1. Do you think it would be fair to include it into the magic methods list?
2. If not, it could perhaps be handy to have a separate rule to ignore missing docstrings in dunder methods?

**Comment by sigmavirus24** [comment]
`__call__` is absolutely a public method, I don't see anything wrong with this. Besides your Flake8 ignore doesn't include `D102` so I'm not sure why the ignore list is relevant for this.

**Comment by AlexArcPy** [comment]
Thanks for the post. My point was that `__repr__`, e.g., is a magic method, but `__call__` is not according to `pydocstyle`. Don't you think it make sense to merge them into the same category of magic methods?

**Comment by sigmavirus24** [comment]
But `__call__` implements a public interface. When you do `f = Foo(), f(1, 2, 3,4)` that's a public interface. `__repr__` is a implementation detail that makes sense to be considered "magic".   `__str__`, `__len__`, etc. are all "magic" because they power things like `str()`, `len()`, etc.

**Comment by mentalisttraceur** [comment]
@AlexArcPy `__call__` is a very *unique* magic method, and in particular it needs documentation a lot more than magic methods like `__repr__` do.

That is why it's not part of the same disabling code. Because people *should* make the decision to not document something like `__call__` (or `__init__`, or `__new__`) separately from the decision to not document all the other magic methods.

Why? Because if I tell you the class defines a `__str__` or whatever, you can trust that you know everything you need to know about that method. But you *don't* know everything you might need to know about a `__call__` method just from the name.

That's why it's a qualitatively different thing, and is not covered by D105.

**Comment by AlexArcPy** [comment]
Ah many thanks @mentalisttraceur that was a helpful explanation, I am closing the issue!
